WordPress/gutenberg
# Fix Legacy Widget block in Customizer

## Description

Fixes https://github.com/WordPress/gutenberg/issues/31094.

- Prevent crash when Legacy Widget block is added to Customizer.
- Show Legacy Widget block variations in Customizer.
- Prevent hidden widget types from appearing in Legacy Widget block dropdown in Customizer.
- Moves common functionality in `edit-widgets` and `customise-widgets` to a new `widgets` package.

## How has this been tested?

1. Enable the Widgets screen in Customizer experiment.
2. Install and activate the SiteOrigin Widgets Bundle plugin.
2. Navigate to Appearance → Customize → Widgets.
3. Add a Legacy Widget block and select a widget. This should now work.
5. Open the inserter and select a SiteOrigin widget variation. This should also now work.

## Checklist:
